### ISSUE TITLE
Feature request: Support getting last N items (backward pagination)

### ISSUE THREAD

**Issue by vhenzl** [paysera/lib-pagination#5]
Thanks for this library, nice to find a cursor pagination implementation for Doctrine!
 
It would be great if it also could support getting _last_ N items (backward pagination), alongside the normal, forward pagination that returns _first_ N items.

The motivation is to use it for [GraphQL Relay pagination](https://relay.dev/graphql/connections.htm) which should support getting both first N items (after the end/next cursor) and last N items (before the start/previous cursor).

From what I have gathered, to implement backward pagination, the queried order has to be flipped (ASC replaced with DESC and vice versa) and returned items reversed internally:
 - https://github.com/graphql/graphql-relay-js/issues/94
 - https://gist.github.com/pcattori/2bb645d587e45c9fdbcabf5cef7a7106

**Comment by mariusbalcytis** [comment]
Yes, this is possible, indeed, just like mentioned in the caveats section of the readme:

> if you need to skip to the last page, you should offer (or do that automatically) to reverse the ordering and go to first page instead;

I'm really not sure if this should be an option done by this library itself, though.
How would you imagine this? Probably some properties / setter in the `Pager` class?

**Comment by vhenzl** [comment]
Hi @mariusbalcytis! Thanks for your response and sorry that mine took so long. I was eventually able to create "proof of concept" of Relay pagination in my code using this library. To get the backwards pagination working, I had to do the following steps:

If `last` arg is set then
1. call `setLimit()` with `last`
2. swap `before` and `after` args when calling `setAfter()` and `setBefore()` (i.e. call `setAfter($before)`)
3. invert the direction of all `OrderingPair`s set to `Pager`
4. reverse the result to get correct order of items
5. swap `Result#getPreviousCursor()` and `Result#getNextCursor()` to get correct `startCursor` and `endCursor` for Relay's page info object
5. swap `Result#hasPrevious()` and `Result#hasNext()` to get correct `hasPreviousPage` and `hasNextPage` for Relay's page info object

All the swapping/reversing is given by a single condition - the existence of `last` argument. 

So I was thinking about having `setFirst()` and `setLast()` method on `Pager`. Existing `setLimit()` would be an alias to `setFirst()`. Calling `setLast()` with non-null value would trigger backwards pagination and applied all the above steps internally so that it would be transparent to the library users.

However, I have encounter other problems with the Relay pagination implementation and I have decided not to use this library for now - so **feel free to close this issue** if you are not interested in backward pagination support.

Anyway, thanks for the library. I may return and give it another try in the future.

**Comment by mariusbalcytis** [comment]
Out of curiosity, what do you mean exactly by Relay pagination implementation? Are those issues related to this library (missing features etc, disregarding last page functionality)?

**Comment by vhenzl** [comment]
It has a prescribed format by [GraphQL Cursor Connections Specification](https://relay.dev/graphql/connections.htm) – connections, edges, page info objects with some mandatory fields like `cursor` on the edge type and additional, conventional fields like `totalCount` on the connection type. See GitHub's [IssueConnection](https://docs.github.com/en/graphql/reference/objects#issueconnection) for a good example.

One of the problems is that I need to be able to create a cursor for each edge (item). But to be able to call `CursorBuilder#getCursorFromItem()` manually I need to somehow obtain an array of `OrderingConfiguration`s. And to do so, I have to analyse the query again and call internal `analyseQuery()` which doesn't seem to be right.

Another of my concerns is about performance. It's common to query just `totalCount` field without `edges` (or `nodes`) on a connection, but the current implementation always loads entities from the DB even if they are not needed. The approached used in Doctrine's [Paginator](https://github.com/doctrine/orm/blob/2.7/lib/Doctrine/ORM/Tools/Pagination/Paginator.php) when data are loaded only if needed seems to be better.

The third thing I noticed is the value of cursors returned from `getPreviousCursor()` and `getNextCursor()` for empty results. I didn't put many thoughts to this but my expectation is they should always return `null` for the empty result but sometimes they return this "with inclusion" ("=") cursors instead.
